src/map/mob.cpp

```diff
@@ -5931,51 +5922,103 @@ static int mob_read_sqlskilldb(void)
 	return 0;
 }
 
+const std::string MobItemRatioDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/mob_item_ratio.yml";
+}
+
 /**
- * Read mob_item_ratio.txt
+ * Reads and parses an entry from the mob_item_ratio.
+ * @param node: YAML node containing the entry.
+ * @return count of successfully parsed rows
  */
-static bool mob_readdb_itemratio(char* str[], int columns, int current)
-{
-	t_itemid nameid;
-	int ratio, i;
-	struct s_mob_item_drop_ratio *item_ratio;
-	nameid = strtoul(str[0], nullptr, 10);
+uint64 MobItemRatioDatabase::parseBodyNode(const YAML::Node &node) {
+	std::string item_name;
 
-	if (itemdb_exists(nameid) == NULL) {
-		ShowWarning("mob_readdb_itemratio: Invalid item id %u.\n", nameid);
+	if (!this->asString(node, "Item", item_name))
+		return false;
+
+	item_data *item = itemdb_search_aegisname(item_name.c_str());
+
+	if (item == nullptr) {
+		this->invalidWarning(node["Item"], "Item %s does not exist, skipping.\n", item_name.c_str());
 		return false;
 	}
 
-	ratio = atoi(str[1]);
+	t_itemid nameid = item->nameid;
 
-	if (!(item_ratio = (struct s_mob_item_drop_ratio *)uidb_get(mob_item_drop_ratio,nameid)))
-		CREATE(item_ratio, struct s_mob_item_drop_ratio, 1);
+	std::shared_ptr<s_mob_item_drop_ratio> data = this->find(nameid);
+	bool exists = data != nullptr;
 
-	item_ratio->drop_ratio = ratio;
-	memset(item_ratio->mob_id, 0, sizeof(item_ratio->mob_id));
-	for (i = 0; i < columns-2; i++) {
-		uint16 mob_id = atoi(str[i+2]);
-		if (mob_db.find(mob_id) == nullptr)
-			ShowError("mob_readdb_itemratio: Invalid monster with ID %hu (Item:%u Col:%d).\n", mob_id, nameid, columns);
-		else
-			item_ratio->mob_id[i] = atoi(str[i+2]);
+	if (!exists) {
+		if (!this->nodesExist(node, { "Ratio" })) {
+			return 0;
+		}
+
+		data = std::make_shared<s_mob_item_drop_ratio>();
+		data->nameid = nameid;
 	}
+	
+	if (this->nodeExists(node, "Ratio")) {
+		uint16 ratio;
+
+		if (!this->asUInt16(node, "Ratio", ratio))
+			return 0;
 
-	if (!item_ratio->nameid) {
-		item_ratio->nameid = nameid;
-		uidb_put(mob_item_drop_ratio, nameid, item_ratio);
+		data->drop_ratio = ratio;
 	}
 
-	return true;
-}
+	if (this->nodeExists(node, "List")) {
+		const YAML::Node &MobNode = node["List"];
 
-/**
- * Free drop ratio data
- **/
-static int mob_item_drop_ratio_free(DBKey key, DBData *data, va_list ap) {
-	struct s_mob_item_drop_ratio *item_ratio = (struct s_mob_item_drop_ratio *)db_data2ptr(data);
-	aFree(item_ratio);
-	return 0;
+		for (const YAML::Node &mobit : MobNode) {
+			if (this->nodeExists(mobit, "Clear")) {
+				std::string mob_name;
+
+				if (!this->asString(mobit, "Clear", mob_name))
+					continue;
+
+				std::shared_ptr<s_mob_db> mob = mobdb_search_aegisname(mob_name.c_str());
+
+				if (mob == nullptr) {
+					this->invalidWarning(mobit["Clear"], "Unknown mob %s, skipping.\n", mob_name.c_str());
+					continue;
+				}
+				uint16 mob_id = mob->vd.class_;
```

Comment:
```suggestion
				uint16 mob_id = mob->id;
```
This was added to master lately.